substrate/frame/scheduler/src/lib.rs

```diff
@@ -440,6 +462,86 @@ pub mod pallet {
 			)?;
 			Ok(())
 		}
+
+		/// Set a retry configuration for a task so that, in case its scheduled run fails, it will
+		/// be retried after `period` blocks, for a total amount of `retries` retries or until it
+		/// succeeds.
+		///
+		/// If a retried task runs successfully before running out of retries, its remaining retry
+		/// counter will be reset to the initial value. If a retried task runs out of retries, it
+		/// will be removed from the schedule.
+		///
+		/// Tasks which need to be scheduled for a retry are still subject to weight metering and
+		/// agenda space, same as a regular task. Periodic tasks will have their periodic schedule
+		/// put on hold while the task is retrying.
+		#[pallet::call_index(6)]
+		#[pallet::weight(<T as Config>::WeightInfo::set_retry(T::MaxScheduledPerBlock::get()))]
+		pub fn set_retry(
+			origin: OriginFor<T>,
+			when: BlockNumberFor<T>,
+			index: u32,
+			retries: u8,
+			period: BlockNumberFor<T>,
+		) -> DispatchResult {
+			T::ScheduleOrigin::ensure_origin(origin.clone())?;
+			let origin = <T as Config>::RuntimeOrigin::from(origin);
+			let agenda = Agenda::<T>::get(when);
+			match agenda.get(index as usize) {
```

Comment:
You can also do something like this `let task = agenda.get(index as usize).flatten().ok_ok(NotFound)?;`.

Reply:
It would need an additional `map` because of the borrow and it would look like:

```
let task = agenda
	.get(index as usize)
	.map(Option::as_ref)
	.flatten()
	.ok_or(Error::<T>::NotFound)?;
```

I think this is a tiny bit uglier, but it might be better because we wouldn't be using the wildcard in the `match`. Let me know what you think.

Reply:
This works, but not that big of a difference:  
```rust
let Some(task) = agenda.get(index as usize).and_then(|x| x.as_ref()) else {
	return Err(Error::<T>::NotFound.into())
};
	
if matches!(
	T::OriginPrivilegeCmp::cmp_privilege(origin.caller(), &task.origin),
	Some(Ordering::Less) | None
) {
	return Err(BadOrigin.into())
}

Reply:
I went with the one above.